### PR TITLE
[DBEX] transform and submit generated 0781v2 PDF

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

- *This work is behind a feature toggle (flipper): ~YES/NO~* **not directly**
  - In this commit, a pre-set [disability_compensation_sync_modern_0781_flow](https://api.va.gov/flipper/features#:~:text=disability_compensation_sync_modern_0781_flow) value determines which form content data are translated when a submission is created
- *(Summarize the changes that have been made to the platform)*
  - leverage new JSON to PDF upload service implemented in https://github.com/department-of-veterans-affairs/vets-api/pull/19802 
  - translate form content to submission data based on form version as indicated by toggle value
  - utilize the presence of 'form0781v2' in a submission's form content when processing a 0781 form
  - create fixture json and pdf files to test all varieties of form content, modeled after pre-existing tests for the current 0781 and 0781a forms
  - rename PDF file name to match expected
- *(If bug, how to reproduce)* **N/A**
- *(What is the solution, why is this the solution?)*
  - The current 21-0781 form that is generated when completing the online form 21-526EZ is outdated (expired 7/21/2020) and requires an upgrade to be in sync with the current paper form (released this summer). The update requires conditional logic to support both the current and new versions simultaneously.
- *(Which team do you work for, does your team own the maintenance of this component?)*
  - Disability Benefits Experience Team 2 (dBeX Carbs 🥖), which owns maintenance of all files in this PR- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/98339

## Testing done

- [X] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
  - modified the following specs to account for and test the validity of new JSON and PDF content 
     - [spec/sidekiq/evss/disability_compensation_form/submit_form0781_spec.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/4ef0021d74566b28b2d788ae6fa8917b2df3907e/spec/sidekiq/evss/disability_compensation_form/submit_form0781_spec.rb)
    - [spec/lib/evss/disability_compensation_form/form0781_spec.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/4ef0021d74566b28b2d788ae6fa8917b2df3907e/spec/lib/evss/disability_compensation_form/form0781_spec.rb)
- *If this work is behind a flipper:* **N/A**

## What areas of the site does it impact?

- translation of and creation of form content data that fills form 21-0781

## Acceptance criteria

- [X]  I ~fixed~|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [X]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/evss/disability_compensation_form/form0781.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def initialize(user, form_content)
         @user = user
         @phone_email = form_content.dig('form526', 'phoneAndEmail')
         @form_content = form_content.dig('form526', 'form0781')
+        @sync_modern_0781_flow = form_content.dig('form526', 'syncModern0781Flow')
```
Comment (kylesoskin) [inline lib/evss/disability_compensation_form/form0781.rb:15]: Are we not able to determine based on the structure such that we need to add this key to the overall 526 submission data? Like that `'form0781v2'` key from above?

File: lib/evss/disability_compensation_form/form0781.rb [hunk 2]
```diff
@@ -47,11 +77,7 @@ def create_form(incidents)
           'email' => @phone_email['emailAddress'],
           'veteranPhone' => @phone_email['primaryPhone'],
           'veteranSecondaryPhone' => '', # No secondary phone available in 526 PreFill
-          'veteranServiceNumber' => '', # No veteran service number available in 526 PreFill
-          'incidents' => incidents,
-          'remarks' => @form_content['remarks'],
-          'additionalIncidentText' => @form_content['additionalIncidentText'],
-          'otherInformation' => @form_content['otherInformation']
+          'veteranServiceNumber' => '' # No veteran service number available in 526 PreFill
```
Comment (kylesoskin) [inline lib/evss/disability_compensation_form/form0781.rb:80]: I am not sure if this is a number we have somewhere (or can get via api call). If it is not, are we asking them for it? Or just leaving it blank on the form? Is it on the form?

**Review by kylesoskin — APPROVED** [review]

File: lib/evss/disability_compensation_form/form0781.rb (hunk 1 above)
Comment (freeheeling) [reply lib/evss/disability_compensation_form/form0781.rb:15]: Do you mean the `'form0781'` key on L14?

Within `'form0781'`, I wasn't able to determine a better attribute that would identify the form content contains data corresponding to the new 0781 form. The `'syncModern0781Flow'` attribute will be submitted as part of the `'form526'` object, regardless, as it persists throughout the form flow to determine which pages are rendered. Therefore, I figured that was as good an indicator as any with which to determine form translation.

Would be happy to entertain any suggestions you might have, however.

File: lib/evss/disability_compensation_form/form0781.rb (hunk 2 above)
Comment (freeheeling) [reply lib/evss/disability_compensation_form/form0781.rb:80]: No change was made here from how 0781 translation currently is configured and documented. There is a service number field on both the current and new form. Would you propose a change be made to obtain this value?

File: lib/evss/disability_compensation_form/form0781.rb (hunk 1 above)
Comment (kylesoskin) [reply lib/evss/disability_compensation_form/form0781.rb:15]: You are doing

`form_content = parsed_forms[form_key]`

`form_key` can equal:

- 'form0781' 
- 'form0781a'
- 'form0781v2' 

This implies, there is a `submission.form['form0781']['form0781v2']` in the submission data, when there is new form data. 

Unless I am misunderstanding? I would think, you key off that, then dont need an extra key. The data/key is either there, or not. 

Again, I may be missing something but that was my thought process.

File: lib/evss/disability_compensation_form/form0781.rb (hunk 2 above)
Comment (kylesoskin) [reply lib/evss/disability_compensation_form/form0781.rb:80]: oh I see, I didnt see that it was already up there and was just moved, it looked like you had added it. My bad. No I dont think, if we dont ask for it, then maybe we dont need to? 

Might be a questions for Ruben? IE "hey there is this service number and Secondary phone number on the form, we dont collect that, should we? Do we need to? etc

idk that this is blocking merge of this though! but we should find out.

File: lib/evss/disability_compensation_form/form0781.rb (hunk 1 above)
Comment (freeheeling) [reply lib/evss/disability_compensation_form/form0781.rb:15]: Those 3 keys, 'form0781', 'form0781a' and 'form0781v2', are created in the `translate` method below. 

`form_content` is what is submitted by the FE, which doesn't currently break out 'form0781' and 'form0781a' data by key; it's different that what is transformed here and saved to a submission record. 

The 'form0781' and 'form0781a' data come nested under the 'form0781' key, and are differentiated below in the `split_incidents` method, depending on the value of the `'personalAssault'` attribute for each `incident'`.

The FE-submitted data and JSON schema could be modified to contain all new 0781 data under a 'form0781v2' key, but I thought the approach was to follow the current process.

Let me know your thoughts. Thanks.

**Review by kylesoskin — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]

**Pull request merged**